src/types/state.cairo

```diff
@@ -0,0 +1,10 @@
+//! State is a top level struct containing the chain state and the utxo set
+
+use super::chain_state::ChainState;
+use super::utxo_set::UtxoSet;
+
+#[derive(Default, Destruct)]
+pub struct State {
+    chain_state: ChainState,
+    utxo_set: UtxoSet
```

Comment:
Here we only need UtreexoState (roots + number of leaves)
UtxoSet also contains local cache (which is not part of input/output) + probably some additional validation context